samples/UpgradeStepSample/UpgradeStepSample/AuthorsPropertyUpgradeStep.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.UpgradeAssistant;
+using Microsoft.DotNet.UpgradeAssistant.Extensions;
+using Microsoft.Extensions.Logging;
+
+namespace UpgradeStepSample
+{
+    /// <summary>
+    /// Upgrade steps all derived from the Microsoft.DotNet.UpgradeAssistant.UpgradeStep base class.
+    /// </summary>
+    public class AuthorsPropertyUpgradeStep : UpgradeStep
+    {
+        private const string AuthorsPropertyName = "Authors";
+        private const string AuthorsPropertySectionName = "AuthorsProperty";
+
+        private readonly AuthorsPropertyOptions? _options;
+        private readonly ILogger<AuthorsPropertyUpgradeStep> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthorsPropertyUpgradeStep"/> class.
+        /// An upgrade step's constructor should have an ILogger parameter along with parameters for any other
+        /// services that need resolved from the dependency injection container. In extensions, AggregateExtension
+        /// is a useful service for providing access to extensions' configuration (as read from extension manifests).
+        /// </summary>
+        /// <param name="logger">Used for logging diagnostic messages. Must be passed to the upgrade step's base class ctor.</param>
+        /// <param name="aggregateExtension">A useful service that provides access to extensions' configuration settings.</param>
+        public AuthorsPropertyUpgradeStep(AggregateExtension aggregateExtension, ILogger<AuthorsPropertyUpgradeStep> logger)
+            : base(logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            if (aggregateExtension is null)
+            {
+                throw new ArgumentNullException(nameof(aggregateExtension));
+            }
+
+            // The AggregateExtension type can be used to get files and configuration settings from extensions.
+            _options = aggregateExtension.GetOptions<AuthorsPropertyOptions>(AuthorsPropertySectionName);
+        }
+
+        // ***** ***** *****
+        // Required overrides
+        // ***** ***** *****
+
+        /// <summary>
+        /// Gets a unique ID string used for referring to the upgrade step from other steps.
+        /// </summary>
+        public override string Id => typeof(AuthorsPropertyUpgradeStep).FullName!;
```

Comment:
should we make this be a virtual method and default to GetType().FullName!;?

Reply:
Yea, that makes sense. I'll do that.